core/lib/l1_contract_interface/src/i_executor/structures/commit_batch_info.rs

```diff
@@ -1,32 +1,34 @@
+use once_cell::sync::Lazy;
+use zkevm_test_harness_1_4_2::kzg::KzgSettings;
 use zksync_types::{
     commitment::{pre_boojum_serialize_commitments, serialize_commitments, L1BatchWithMetadata},
     ethabi::Token,
     web3::{contract::Error as Web3ContractError, error::Error as Web3ApiError},
     U256,
 };
 
-use crate::Tokenizable;
+use crate::{
+    i_executor::commit::kzg::{KzgInfo, ZK_SYNC_BYTES_PER_BLOB},
+    Tokenizable,
+};
 
 /// Encoding for `CommitBatchInfo` from `IExecutor.sol`
 #[derive(Debug)]
 pub struct CommitBatchInfo<'a>(pub &'a L1BatchWithMetadata);
 
-impl<'a> Tokenizable for CommitBatchInfo<'a> {
-    fn from_token(_token: Token) -> Result<Self, Web3ContractError>
-    where
-        Self: Sized,
-    {
-        // Currently there is no need to decode this struct.
-        // We still want to implement `Tokenizable` trait for it, so that *once* it's needed
-        // the implementation is provided here and not in some other inconsistent way.
-        Err(Web3ContractError::Api(Web3ApiError::Decoder(
-            "Not implemented".to_string(),
-        )))
-    }
+pub fn load_kzg_settings() -> KzgSettings {
+    static KZG_SETTINGS: Lazy<KzgSettings> = Lazy::new(|| {
+        let zksync_home = std::env::var("ZKSYNC_HOME").unwrap_or_else(|_| ".".into());
+        let path = std::path::Path::new(&zksync_home).join("trusted_setup.json");
+        KzgSettings::new(path.to_str().unwrap())
```

Comment:
```suggestion
        KzgSettings::new(path.to_str().expect("kzg setup file trusted_setup.json is not present at expected location; qed")
```
a panic with a message is a bit more usable